teuthology/provision/maas.py

```diff
@@ -0,0 +1,441 @@
+import json
+import logging
+
+from oauthlib.oauth1 import SIGNATURE_PLAINTEXT
+from requests_oauthlib import OAuth1Session
+
+import teuthology.orchestra
+
+from teuthology.config import config
+from teuthology.contextutil import safe_while
+from teuthology.orchestra.opsys import OS
+from teuthology import misc
+
+log = logging.getLogger(__name__)
+
+ubuntu_codenames = {
+    "20.04": "focal",
+    "22.04": "jammy",
+    "24.04": "noble",
+    "25.04": "plucky",
+    "25.10": "questing",
+}
```

Comment:
I would suggest using `teuthology.orchestra.opsys.DISTRO_CODENAME_MAP` instead of this.